ts/src/bitrue.ts

```diff
@@ -1326,63 +1929,108 @@ export default class bitrue extends Exchange {
          * @param {string} [params.clientOrderId] a unique id for the order, automatically generated if not sent
-         *
+         * @param {decimal|undefined} [params.leverage] in future order, the leverage value of the order should consistent with the user contract configuration
          * EXCHANGE SPECIFIC PARAMETERS
          * @param {decimal} [params.icebergQty]
          * @param {long} [params.recvWindow]
          * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
          */
         await this.loadMarkets ();
         const market = this.market (symbol);
+        let marketType = undefined;
+        [ marketType, params ] = this.handleMarketTypeAndParams ('createOrder', market, params);
+        let subType = undefined;
+        [ subType, params ] = this.handleSubTypeAndParams ('createOrder', market, params);
+        let response = undefined;
+        let data = undefined;
         const uppercaseType = type.toUpperCase ();
-        const validOrderTypes = this.safeValue (market['info'], 'orderTypes');
-        if (!this.inArray (uppercaseType, validOrderTypes)) {
-            throw new InvalidOrder (this.id + ' ' + type + ' is not a valid order type in market ' + symbol);
-        }
         const request = {
-            'symbol': market['id'],
             'side': side.toUpperCase (),
             'type': uppercaseType,
             // 'timeInForce': '',
-            'quantity': this.amountToPrecision (symbol, amount),
             // 'price': this.priceToPrecision (symbol, price),
             // 'newClientOrderId': clientOrderId, // automatically generated if not sent
             // 'stopPrice': this.priceToPrecision (symbol, 'stopPrice'),
             // 'icebergQty': this.amountToPrecision (symbol, icebergQty),
         };
-        const clientOrderId = this.safeString2 (params, 'newClientOrderId', 'clientOrderId');
-        if (clientOrderId !== undefined) {
-            params = this.omit (params, [ 'newClientOrderId', 'clientOrderId' ]);
-            request['newClientOrderId'] = clientOrderId;
-        }
         if (uppercaseType === 'LIMIT') {
             if (price === undefined) {
                 throw new InvalidOrder (this.id + ' createOrder() requires a price argument');
             }
             request['price'] = this.priceToPrecision (symbol, price);
         }
-        const stopPrice = this.safeValue2 (params, 'triggerPrice', 'stopPrice');
-        if (stopPrice !== undefined) {
-            params = this.omit (params, [ 'triggerPrice', 'stopPrice' ]);
-            request['stopPrice'] = this.priceToPrecision (symbol, stopPrice);
+        if (market['swap']) {
+            request['contractName'] = market['id'];
+            request['amount'] = this.parseNumber (amount);
+            request['volume'] = this.parseNumber (amount);
+            request['positionType'] = 1;
+            request['open'] = 'OPEN';
```

Comment:
if reduceOnly is provided shouldn't we set this to false?
![image](https://github.com/ccxt/ccxt/assets/43336371/caf1bb8e-cd64-4dd0-b825-3973e86de844)